caddy.go

```diff
@@ -224,6 +224,10 @@ func changeConfig(method, path string, input []byte, ifMatchHeader string, force
 	idx := make(map[string]string)
 	err = indexConfigObjects(rawCfg[rawConfigKey], "/"+rawConfigKey, idx)
 	if err != nil {
+		err2 := rollbackRawCfg()
+		if err2 != nil {
+			err = errors.Join(err, err2)
+		}
```

Comment:
Trying to parse the old config here without checking the length of `rawCfgJSON` isn't a recommended approach. Additionally, this is a potential duplicate of the call in `Ln246`